### PR TITLE
Sorts imports and fixes modernizes types for Flyout component

### PR DESCRIPTION
**Pull request by rozele**
### Why
Makes a few minor adjustments to Flyout.js and FlyoutNativeComponents.js to satisfy a few stricter flow and lint checks.

## Changelog
Should this change be included in the release notes: _no_
 ###### Microsoft Reviewers: [Open in CodeFlow](https://microsoft.github.io/open-pr/?codeflow=https://github.com/microsoft/react-native-windows/pull/12601)

### CODE REVIEW

File: vnext/src/Libraries/Components/Flyout/Flyout.js [hunk 1]
```diff
@@ -3,14 +3,13 @@
  * Licensed under the MIT License.
  * @format
  */
```
Comment (NickGerleman) [inline vnext/src/Libraries/Components/Flyout/Flyout.js:5]: Should this have an `@flow strict` or `@flow strict-local`?

Comment (jonthysell) [reply vnext/src/Libraries/Components/Flyout/Flyout.js:5]: Probably?

**Review by jonthysell — APPROVED** [review]

**Review by marlenecota — APPROVED** [review]

**Comment by chiaramooney** [comment]
Change is approved. Will merge when CI runs green.

**Comment by chrisglein** [comment]
> Change is approved. Will merge when CI runs green.

@rozele This PR is older but had signoff pending a clean CI. If you can get that running clean, seems good to go?

**Review by marlenecota — APPROVED** [review]

**Pull request merged**